cmd/zarb/tx/bond_tx.go

```diff
@@ -112,10 +110,52 @@ func BondTx() func(c *cli.Cmd) {
 				return
 			}
 
+			//sign transaction
+			if *keyFileOpt == "" {
+				cmd.PrintWarnMsg("Please specify a key file to sign.")
+				c.PrintHelp()
+				return
+			}
+			if *authOpt == "" {
+				auth = cmd.PromptPassphrase("Passphrase: ", false)
+			} else {
+				auth = *authOpt
+			}
+
+			//RPC
+			if *grpcOpt == "" {
+				rpc = cmd.PromptInput("gRPC server address: ")
+			} else {
+				rpc = *grpcOpt
+			}
+
+			grpcClient, err := cmd.GetRPCClient(rpc)
+			if err != nil {
+				cmd.PrintErrorMsg("Couldn't connect to RPC Server: %v", err)
+				return
+			}
+
+			seq = *seqOpt
+			if *seqOpt == 0 {
+				seq, err = cmd.GetSequence(grpcClient, bonder)
```

Comment:
I have a suggestion here. 
If you make `func grpcClient(grpcOpt *string) {... }`
Then you only ask user anytime you want to query or broadcast the network. 
`cmd.GetSequence(grpcClient(grpcOpt), bonder)`

So if user wants to set all the options manually (offline), no need to ask gRPC endpoint.

Reply:
@b00f
I tried to implement it this idea.
let me know if you got better way

Reply:
i tried to implement this idea let me know if you got better way